src/toil/job.py

```diff
@@ -41,55 +42,54 @@
 logger = logging.getLogger( __name__ )
 
 
-class Job(object):
+class ResourceRequirementMixin(object):
     """
-    Class represents a unit of work in toil.
+    Inherit from this class to add resource requirements for an object
     """
-    def __init__(self, memory=None, cores=None, disk=None, preemptable=None, checkpoint=False):
+    def __init__(self, memory=None, cores=None, disk=None):
+        self._cores = self._parseResource('cores', cores)
+        self._memory = self._parseResource('memory', memory)
+        self._disk = self._parseResource('disk', disk)
+        self._config = None
+
+    @property
+    def disk(self):
+        """
+        This property insures that even if the user does not pass a resource requirment, the
+        default value will be available when the user's code is run in Job.run or in
+        Job.Service.start. If a value is not set and the Job is not yet ready to be run, this method
+        will return None.
+        """
+        if self._disk or self._config:
```

Comment:
You're right, it was going to be an error but there is a case in ServiceJob where we refer to the resources without _config set, in which case we want to return None and pass the burden of determining the actual default values into Job. Alternatively, I could make this an error and try-except it in ServiceJob.